Document potential data loss when changing metric types (#780)
mozilla/glean: docs/user/metric-parameters.md

- `type`: **Required.**  Specifies the type of a metric, like "counter" or "event". This defines which operations are valid for the metric, how it is stored and how data analysis tooling displays it. See the list of [supported metric types](metrics/index.md).

> > **Important**: Once a metric is released in a product, its `type` should not be changed. If any data was collected locally with the older `type`, before the ping it goes in has a chance to be submitted, recording data with the new `type` may cause any old persisted data to be lost for that metric. See [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1621757#c1) for an extended explanation of the different scenarios.

Reviewer: ```suggestion
> **Important**: Once a metric is released in a product, its `type` should not be changed. If any data was collected locally with the older `type`, and hasn't yet been sent in a ping, recording data with the new `type` may cause any old persisted data to be lost for that metric. See [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1621757#c1) for an extended explanation of the different scenarios.
```

Reviewer: Slightly clearer to me, maybe, but not incorrect either way.
